src/services/rfqm_service.ts

```diff
@@ -28,6 +30,29 @@ export interface FetchIndicativeQuoteResponse {
     sellAmount: BigNumber;
     sellTokenAddress: string;
 }
+export interface FetchFirmQuoteParams {
+    apiKey: string;
+    buyAmount?: BigNumber;
+    buyToken: string;
+    buyTokenDecimals: number;
+    gasPrice?: BigNumber;
+    sellAmount?: BigNumber;
+    sellToken: string;
+    sellTokenDecimals: number;
+    takerAddress: string;
+}
+
+export interface FetchFirmQuoteResponse {
```

Comment:
I think we should make this more generic, as we may want to change the order format.

Instead of `metaTransaction` I would propose a `payload` key, that could be a MetaTransaction object but could be something different, like an order to sign.

I think we should return the order hash as well, and the `metaTranactionHash` field could be optional. It might even be worth adding a new key called `identifiers` which could look like:

```
identifiers: {
    orderHash: string;
    metaTransactionHash?: string;
}
```

Reply:
This is a really good idea - will noodle on this a bit. I think there's two "flavors" of RFQ "M" to consider where:

- M = metatransaction, where the metatransaction needs to be signed
- M = maker, where the roles are flipped vs RFQT, and as you mentioned, an order needs to be signed

The identifiers also does make a lot of sense. I like that too.